### DISCUSSION TITLE
Tunny v0.11.0

### DISCUSSION THREAD

**Discussion by hrntsm** [Tunny-gh/Tunny_in_Open_Sea#282] [Announcements]
## Release note

[リリースノート](https://graceful-stag-dae.notion.site/Tunny-Release-Note-v0-11-56fd89c310164e8887e09384e00e920e)
[Release Note](https://graceful-stag-dae.notion.site/Tunny-Release-Note-v0-11-125bbf8f18e04588a4700d0d1f199fd4)

## Added

- Bone Fish component
  - It is Tunny UI less mode component.
- Categorical optimization.
  - Sample gh file also added
- Log output.
  - Improved ease of support.
- Log level selector.
  - It is located in the Misc tab of the Settings tab.
  - There are three levels: "Verbose," "Debug," and "Information.
- New GP sampler support
  - This is related optuna 3.6
- Sample python code to create a gif animation.
- Tunny.Core csproj for improve develop environment.
- CI build with python.

## Changed

- DeconstructFish component output
  - To support categorical optimization, the variable output was split from "Variables" into "NumberVariables" and "TextVariables".
- Move FishPrint component to Print subcategory
- Move Construct & Deconstruct component ot Operation subcategory
- Use optuna.csproj sqlite handler instead of tunny.storage
- Improve result fish output to use new storage handler
  - The handler is independent from python process.
- Human in the loop mode support log storage format.
- Allows clustering with any combination of variables and objective function.
- Speedup optimization.
  - Do not recalculate irrelevant components.
  - Previously, all components were recalculated for each trial, but this has been changed so that components not involved in optimization are not recalculated
- Bump Python library
  - optuna-dashboard 0.14 to 0.15
    - csv can be downloaded from the trial table.
    - The Pareto Front and History plots can jump to the corresponding trial page when click dot plot.
    - Settings have been added to change the appearance of the plots.
    - When uploading surfaces with no thickness to Artifacts, both sides are rendered.
    - Human-in-the-loop works with journal storage
    - Fix overflow when preferential optimization
    - [see more detail](https://github.com/optuna/optuna-dashboard/releases/tag/v0.15.0)
  - optuna 3.5 to 3.6
    - New GP sampler support
    - Fixed bug in parallelization in log files
    - [see more detail](https://github.com/optuna/optuna/releases/tag/v3.6.0)

## Fixed

- When launching a window, if all the objectives are null, the window will not launch.
- Empty 3dm is always saved even if you want to save only images in artifact.
- FishEgg feature does not work.
- Fish Print by capture timing
  - Viewport capture at the end of all Grasshopper processing.

## Related PR

* Feature logger by @hrntsm in https://github.com/hrntsm/Tunny/pull/232
* Support categorical optimization by @hrntsm in https://github.com/hrntsm/Tunny/pull/233
* Feature/add all method to debug log by @hrntsm in https://github.com/hrntsm/Tunny/pull/237
* Fix to run python install when new tunny version by @hrntsm in https://github.com/hrntsm/Tunny/pull/238
* Fix SaveAllArtifacts method to conditionally save artifacts by @hrntsm in https://github.com/hrntsm/Tunny/pull/239
* Feature/add git animation python sample by @hrntsm in https://github.com/hrntsm/Tunny/pull/240
* Feature/add tunny.core proj by @hrntsm in https://github.com/hrntsm/Tunny/pull/241
* Add Bone Fish component which Tunny UI less component by @hrntsm in https://github.com/hrntsm/Tunny/pull/242
* Feature/codecov by @hrntsm in https://github.com/hrntsm/Tunny/pull/243
* Feature/add sqlite handler in optuna csproj by @hrntsm in https://github.com/hrntsm/Tunny/pull/244
* Update BoneFish to speedup result output by @hrntsm in https://github.com/hrntsm/Tunny/pull/245
* Fix Tunny optimization environment check by @hrntsm in https://github.com/hrntsm/Tunny/pull/246
* Moving Rhino independent files to Tunny.Core by @hrntsm in https://github.com/hrntsm/Tunny/pull/250
* Feature/improve artifact handling by @hrntsm in https://github.com/hrntsm/Tunny/pull/253
* Feature/clean algorithm.cs by @hrntsm in https://github.com/hrntsm/Tunny/pull/254
* Feature/clean visualization by @hrntsm in https://github.com/hrntsm/Tunny/pull/255
* Feature/improve output speed by @hrntsm in https://github.com/hrntsm/Tunny/pull/256
* Fix/fish egg does not run by @hrntsm in https://github.com/hrntsm/Tunny/pull/258
* Fix SO optimization continue error by @hrntsm in https://github.com/hrntsm/Tunny/pull/259
* Fix temp file delete error by @hrntsm in https://github.com/hrntsm/Tunny/pull/261
* Feature/update sample by @hrntsm in https://github.com/hrntsm/Tunny/pull/262
* Refactor output handling in Tunny.Handler.OutputLoop.cs and Tunny.Sol… by @hrntsm in https://github.com/hrntsm/Tunny/pull/263
* Feature/refactor output by @hrntsm in https://github.com/hrntsm/Tunny/pull/264
* Fix/fishprint order by @hrntsm in https://github.com/hrntsm/Tunny/pull/266
* Feature/ci python lib setup by @hrntsm in https://github.com/hrntsm/Tunny/pull/269
* Feature/update optuna dashboard to 0.15 by @hrntsm in https://github.com/hrntsm/Tunny/pull/270
* Fix example file by @hrntsm in https://github.com/hrntsm/Tunny/pull/271
* Feature/update issue form by @hrntsm in https://github.com/hrntsm/Tunny/pull/272
* Feature/do not recalculate irrelevant components by @hrntsm in https://github.com/hrntsm/Tunny/pull/273
* Feature/update optuna 3.6 by @hrntsm in https://github.com/hrntsm/Tunny/pull/274
* Update UITunnyMessages.cs by @hrntsm in https://github.com/hrntsm/Tunny/pull/275
* Feature/update clustering by @hrntsm in https://github.com/hrntsm/Tunny/pull/276
* Fix/fish print by capture expire by @hrntsm in https://github.com/hrntsm/Tunny/pull/277
* Move Construct & DeCon to Operation dir by @hrntsm in https://github.com/hrntsm/Tunny/pull/278
* Bump the version up to v0.11 by @hrntsm in https://github.com/hrntsm/Tunny/pull/279
* Bump the version up to v0.11 by @hrntsm in https://github.com/hrntsm/Tunny/pull/280

**Full Changelog**: https://github.com/hrntsm/Tunny/compare/v0.10.0...v0.11.0

<hr /><em>This discussion was created from the release <a href='https://github.com/hrntsm/Tunny/releases/tag/v0.11.0'>Tunny v0.11.0</a>.</em>

**Comment by SpaceArchitecture** [comment]
This is great, you are by far the best optimization plugin for Grasshopper

**Reply by hrntsm** [reply]
Thanks!
